app/models/spree/adjustment_decorator.rb

```diff
@@ -25,7 +25,7 @@ def self.prepended(base)
       private
 
       def avatax?
-        tax? && self.source.avatax?
+        tax? && self.source && self.source.avatax?
```

Comment:
would the safe operator here save us from using another set of `&&`s ? Something like:

`tax? && self.source&.avatax?`?

Reply:
What is this syntax? Is that like a shorthand for `try`?

Reply:
I did not know that was a feature of the language. Neat. I had considered using a null-object here so I'd have a consistent interface, but the safe-operator is a good compromise. Thanks for the nudge.

Reply:
After some Slack discussion, I'm going to forego this suggestion so I can avoid bumping the required Ruby version.